Source/Utilities/EventBusMixin.lua

```diff
@@ -0,0 +1,41 @@
+AuctionatorEventBusMixin = {}
+
+function AuctionatorEventBusMixin:Init()
+  self.registeredFrames = {}
+  self.sources = {}
+end
+
+function AuctionatorEventBusMixin:Register(listener, eventNames)
+  if listener == nil or listener.frame.EventUpdate == nil then
+    error("Attempted to register an invalid frame! EventUpdate function must be defined.")
+    return
+  end
+
+  for _, eventName in ipairs(eventNames) do
+    if self.registeredFrames[eventName] == nil then
+      self.registeredFrames[eventName] = {}
+    end
+
+    table.insert(self.registeredFrames[eventName], frame)
+  end
+end
+
+function AuctionatorEventBusMixin:RegisterSource(source, name)
+  self.sources[tostring(source)] = name
+end
+
+function AuctionatorEventBusMixin:Fire(source, eventName, eventData)
+  Auctionator.Debug.Message("AuctionatorEventBus:Fire()", eventName, eventData)
+
+  if self.sources[tostring(source)] == nil then
+    error("All sources must be registered")
+  end
+
+  if self.registeredFrames == nil or self.registeredFrames[eventName] == nil then
```

Comment:
We should be able to remove the `self.registeredFrames == nil` check since the event bus is now initialized.